netComparator/cmd/main.go

```diff
@@ -199,51 +192,45 @@ func compareTransactions(txHashes []wrappedTxHashes, retryConfig []retry.Option)
 
 	for i, t := range txHashes {
 		wg.Add(1)
-		go func(i int, t string) {
-			defer wg.Done()
-			var (
-				txM *data.TransactionOnNetwork
-				txS *data.TransactionOnNetwork
-			)
-
-			// Get transaction from shadow-fork in a retry loop.
-			err := retry.Do(
-				func() error {
-					var (
-						err error
-						wd  *wrappedDifferences
-					)
-					txM, wd, err = getTransaction(t, "mainnet", primaryProxy)
-					if err != nil {
-						return err
-					}
-
-					txS, wd, err = getTransaction(t, "shadow-fork", secondaryProxy)
-					if err != nil {
-						return err
-					}
-
-					if wd != nil {
-						wrappedDiffs[i] = *wd
-						return nil
-					}
-
-					wrappedDiffs[i] = getDifference(t, *txM, *txS)
-					return nil
-
-				}, retryConfig...,
-			)
-
-			if err != nil {
-				log.Error(err.Error())
-			}
-		}(i, t.TxHash)
+		compareTransaction(wrappedDiffs, i, t.TxHash, &wg, retryConfig)
 	}
 
 	wg.Wait()
 	return wrappedDiffs
 }
 
+func compareTransaction(wrappedDiffs []wrappedDifferences, i int, t string, wg *sync.WaitGroup, retryConfig []retry.Option) {
+	defer wg.Done()
+
+	// Get transaction from shadow-fork in a retry loop.
```

Comment:
"Get transactions from both the primary and the secondary proxies, in a retry loop" - or something similar.